.pre-commit-config.yaml

```diff
@@ -0,0 +1,206 @@
+      - id: darker
+        args:
+          - --isort
+          - -L
+          - flake8
+        additional_dependencies:
+          - isort==5.10.1
+          - flake8==4.0.1
+          - flake8-bugbear==16.4.0
+          - flake8-import-order==0.18.1
+          - flake8-builtins==1.5.3
+          - flake8-future-annotations==0.0.5
+          - flake8-docstrings==1.6.0
+          - flake8-rst-docstrings==0.2.6
+          - pep8-naming==0.13.0
+
+  # Format changed lines only of C/C++ files
+  # We want to use git-clang-format which is provided by the clang-format wheel,
+  # but not exposed in its precommit hook.
+  # Do not specify args here; use the tools/precommit-clang-format.sh that this
+  # calls.
+  - repo: local
+    hooks:
+      - id: clang-format-diff
+        name: clang-format-diff
+        entry: python tools/precommit-clang-format.py
+        language: python        # Script is bash but need python deps
+        require_serial: true    # Lock errors if it tries to chunk the files
+        types_or: [c++, c]
+        verbose: true
+        additional_dependencies:
+          - clang-format==14.0.6
+
+  # YAML Formatter. Only enabled on one file for now, will add more eventually
+  - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
+    rev: d0cf8a7a8e09127a5a61c0644a823544e32f492b  # frozen: 0.2.2
+    hooks:
+      - id: yamlfmt
+        args: [--mapping, '2', --sequence, '4', --offset, '2', --width, '88']
+        # Will eventually apply to all YAML files
+        files: \.pre-commit-config.yaml
+
+
+  ### Manually-run hooks ###
+  # These hooks are disabled by default because NumPy currently has one or more
+  # files that are known to fail. You are encouraged to run these hooks if you
+  # are changing some files, and aim to have them pass. However, the goal is
+  # that somebody doing development is not required to work on anything outside
+  # of their own section of interest, so do not worry if other parts of files fail.
+  # (though in general, if you would _like_ to fix them, that would be amazing!)
+  # A reason each hook is disabled accompanies each of them
+
+  # Allow black to be run on entire file manually
+  # Disabled for now to avoid whole-file commit diffs
+  # Run with: pre-commit run --hook-stage manual black
+  - repo: https://github.com/psf/black
+    rev: f6c139c5215ce04fd3e73a900f1372942d58eca0  # frozen: 22.6.0
+    hooks:
+      - id: black
+        stages: [manual]
+
+  # Apply black formatting to docstrings and .rst files Disabled for now to
+  # avoid whole-file commit diffs, diff-only formatter is applied by the
+  # "darker" hook
+  #
+  # NOTE: This hook throws a ton of errors. Most of them are because `..
+  # code-block:: python` is used where `.. code-block:: pycon` should be.
+  # `python` is for code blocks as one might see in a source file. `pycon` is
+  # the correct code block type for python console syntax, i.e., blocks with
+  # `>>>`/`...`, like `doctest` might run
```

Comment:
Could you open a documentation issue about this and refer to it? Otherwise this might get lost